internal/awsdata/ecs.go

```diff
@@ -0,0 +1,163 @@
+
+		for !done {
+			outListTasks, err := ecsSvc.ListTasks(params)
+			if err != nil {
+				d.results <- result{Err: err}
+				return
+			}
+
+			taskArns = append(taskArns, outListTasks.TaskArns...)
+
+			if outListTasks.NextToken == nil {
+				done = true
+			} else {
+				params.NextToken = outListTasks.NextToken
+			}
+		}
+
+		// TODO: API call can only handle 100 task ARNs at a time
+		outDescribeTasks, err := ecsSvc.DescribeTasks(&ecs.DescribeTasksInput{
+			Cluster: cluster.ClusterArn,
+			Tasks: taskArns,
+		})
+		if err != nil {
+			d.results <- result{Err: err}
+			return
+		}
+		for _, task := range outDescribeTasks.Tasks {
+			for _, container := range task.Containers {
+				d.wg.Add(1)
+				go d.processECSContainer(container, task, cluster, region)
+			}
+		}
+	}
+
+	log.Info("finished processing instance data")
+}
+
+func (d *AWSData) processECSContainer(container *ecs.Container, task *ecs.Task, cluster *ecs.Cluster, region string) {
+	defer d.wg.Done()
+
+	var ips []string
+	var macAddresses []string
+	var networkInterfaces []string
+
+	for _, attachment := range task.Attachments {
+		if aws.StringValue(attachment.Type) != "ElasticNetworkInterface" {
+			continue
+		}
+		for _, details := range attachment.Details {
+			switch aws.StringValue(details.Name) {
+			case "privateIPv4Address":
+			  ips = append(ips, aws.StringValue(details.Value))
+			case "ipv6Address":
+			  ips = append(ips, aws.StringValue(details.Value))
+			case "macAddress":
+			  macAddresses = append(macAddresses, aws.StringValue(details.Value))
+			case "networkInterfaceId":
+			  networkInterfaces = append(networkInterfaces, aws.StringValue(details.Value))
+			}
+		}
+	}
+
+	for _, networkInterface := range container.NetworkInterfaces {
+		ips = appendIfMissing(ips, aws.StringValue(networkInterface.PrivateIpv4Address))
+		ips = appendIfMissing(ips, aws.StringValue(networkInterface.Ipv6Address))
+	}
+
+	var hardware = aws.StringValue(task.LaunchType)
+	if hardware == "FARGATE" {
+		hardware = fmt.Sprintf("%s %s", hardware, aws.StringValue(task.PlatformVersion))
+	}
+
+	d.results <- result{
+		Row: inventory.Row{
+			UniqueAssetIdentifier:     fmt.Sprintf("%s-%s", aws.StringValue(container.Name), aws.StringValue(container.RuntimeId)),
+			IPv4orIPv6Address:         strings.Join(ips, "\n"),
+			Virtual:                   true,
+			MACAddress:                strings.Join(macAddresses, "\n"),
+			BaselineConfigurationName: aws.StringValue(container.Image),
+			Location:                  region,
+			AssetType:                 AssetTypeECSContainer,
+			HardwareMakeModel:         hardware,
+			Function:                  fmt.Sprintf("%s %s", aws.StringValue(cluster.ClusterName), aws.StringValue(task.Group)),
+			SerialAssetTagNumber:      aws.StringValue(container.ContainerArn),
+			// TODO: Query EC2 API to map ENI to VPC
+			VLANNetworkID:             strings.Join(networkInterfaces, "\n"),
+		},
+	}
+}
+
+func appendIfMissing(slice []string, s string) []string {
```

Comment:
Might be useful for other resources. Any particular place I should put this helper function?

Reply:
Maybe with the other helpers in `data.go`? Either that or a new `util.go`/`helper.go` file. I'll leave that decision up to you

Reply:
I'll probably put it in `data.go` for now to reduce complexity, but should refactor into a separate util/helper at some future point.

Reply:
Yea, looking at the size of AWSData is making me think the whole thing could be refactored into a more modular setup but that's tomorrows problem.